sql/resource_alert.go

```diff
@@ -0,0 +1,121 @@
+package sql
+
+import (
+	"context"
+	"log"
+	"strings"
+
+	"github.com/databricks/databricks-sdk-go/service/sql"
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func ResourceAlert() common.Resource {
+	s := common.StructToSchema(sql.Alert{}, func(m map[string]*schema.Schema) map[string]*schema.Schema {
+		common.CustomizeSchemaPath(m, "display_name").SetRequired()
+		common.CustomizeSchemaPath(m, "query_id").SetRequired()
+		common.CustomizeSchemaPath(m, "condition").SetRequired()
+		// TODO: can we automatically generate it from SDK? Or should we avoid validation at all?
+		common.CustomizeSchemaPath(m, "condition", "op").SetRequired().SetValidateFunc(validation.StringInSlice([]string{
```

Comment:
Yes, we should generate this in the SDK. We should make this a feature request against the SDK if there isn't already one.

Reply:
I didn't find any, filed https://github.com/databricks/databricks-sdk-go/issues/1063 - it's possible to get something with `reflect`, but it's better to have in SDK.